### PR TITLE
Multi-arch images

### PR DESCRIPTION
**Pull request by amulobaev**

### CODE REVIEW

File: full/Dockerfile.arm64 [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+ENV APACHE_DOCROOT_IN_REPO="public"
+ENV NGINX_DOCROOT_IN_REPO="public"
+
+### Homebrew ###
+LABEL dazzle/layer=tool-brew
+LABEL dazzle/test=tests/tool-brew.yaml
+USER gitpod
+# Dazzle does not rebuild a layer until one of its lines are changed. Increase this counter to rebuild this layer.
+ENV TRIGGER_BREW_REBUILD=1
+#RUN mkdir ~/.cache && /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
Comment (pavan-tri) [inline full/Dockerfile.arm64:68]: mkdir -p /home/linuxbrew && cd /home && curl -L https://github.com/Homebrew/brew/tarball/master | tar xz --strip 1 -C linuxbrew

**Comment by alexander-a-chernyshev** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

Please consider using [Analyzer UI](https://confluence.devfactory.com/display/CodeReview/Analyzer+UI) extension to catch issues in pull requests.

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: full/Dockerfile [hunk 2]
```diff
@@ -207,6 +208,20 @@ ENV PATH=$PATH:$HOME/.cargo/bin
 
 RUN bash -lc "cargo install cargo-watch cargo-edit cargo-tree"
 
+### Install additional components ###
+USER gitpod
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
+    unzip awscliv2.zip && sudo ./aws/install
+RUN curl --silent --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_Linux_amd64.tar.gz" | tar xz -C /tmp && \
```
Comment (alexander-a-chernyshev) [inline full/Dockerfile:215]: <details>
<summary>:no_entry_sign: :repeat: <strong>This RUN command is calling sudo.</strong> (click for details)</summary>
First occurrence of this finding in this pull request can be found <a href="https://github.com/trilogy-group/workspace-images/pull/1#discussion_null">here</a></details>

[//]: # (<a id="0463df00-d276-11eb-a888-3141ef71e4d9" data-business-id="9f8ab984-54c1-4ccc-b774-8d8c206bdce5" href="none" data-finding-id="476566751" repeated-comment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+ENV NGINX_DOCROOT_IN_REPO="public"
+
+### Homebrew ###
+LABEL dazzle/layer=tool-brew
+LABEL dazzle/test=tests/tool-brew.yaml
+USER gitpod
+# Dazzle does not rebuild a layer until one of its lines are changed. Increase this counter to rebuild this layer.
+ENV TRIGGER_BREW_REBUILD=2
+#RUN mkdir ~/.cache && /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+RUN sudo mkdir -p /home/linuxbrew/.linuxbrew && cd /home/linuxbrew && sudo curl -L https://github.com/Homebrew/brew/tarball/master | sudo tar xz --strip 1 -C .linuxbrew && sudo chown -R gitpod:gitpod /home/linuxbrew/.linuxbrew
```
Comment (alexander-a-chernyshev) [inline full/Dockerfile.arm64:69]: <details>
<summary>:no_entry_sign: :repeat: <strong>This RUN command is calling sudo.</strong> (click for details)</summary>
First occurrence of this finding in this pull request can be found <a href="https://github.com/trilogy-group/workspace-images/pull/1#discussion_null">here</a></details>

[//]: # (<a id="0e212840-d276-11eb-a888-3141ef71e4d9" data-business-id="9f8ab984-54c1-4ccc-b774-8d8c206bdce5" href="none" data-finding-id="476566751" repeated-comment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+### Go ###
+LABEL dazzle/layer=lang-go
+LABEL dazzle/test=tests/lang-go.yaml
+USER gitpod
+ENV GO_VERSION=1.16.5
+ENV GOPATH=$HOME/go-packages
+ENV GOROOT=$HOME/go
+ENV PATH=$GOROOT/bin:$GOPATH/bin:$PATH
+RUN curl -fsSL https://storage.googleapis.com/golang/go$GO_VERSION.linux-arm64.tar.gz | tar xzs && \
```
Comment (alexander-a-chernyshev) [inline full/Dockerfile.arm64:85]: <details>
<summary>:no_entry_sign: :repeat: <strong>This RUN command is calling sudo.</strong> (click for details)</summary>
First occurrence of this finding in this pull request can be found <a href="https://github.com/trilogy-group/workspace-images/pull/1#discussion_null">here</a></details>

[//]: # (<a id="114dfe30-d276-11eb-a888-3141ef71e4d9" data-business-id="9f8ab984-54c1-4ccc-b774-8d8c206bdce5" href="none" data-finding-id="476566751" repeated-comment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+    && echo ". ~/.nvm/nvm-lazy.sh"  >> /home/gitpod/.bashrc.d/50-node
+# above, we are adding the lazy nvm init to .bashrc, because one is executed on interactive shells, the other for non-interactive shells (e.g. plugin-host)
+COPY --chown=gitpod:gitpod nvm-lazy.sh /home/gitpod/.nvm/nvm-lazy.sh
+ENV PATH=$PATH:/home/gitpod/.nvm/versions/node/v${NODE_VERSION}/bin
+
+### Python ###
+LABEL dazzle/layer=lang-python
+LABEL dazzle/test=tests/lang-python.yaml
+USER gitpod
+RUN sudo install-packages python3-pip
```
Comment (alexander-a-chernyshev) [inline full/Dockerfile.arm64:157]: <details>
<summary>:no_entry_sign: :repeat: <strong>This RUN command is calling sudo.</strong> (click for details)</summary>
First occurrence of this finding in this pull request can be found <a href="https://github.com/trilogy-group/workspace-images/pull/1#discussion_null">here</a></details>

[//]: # (<a id="13e4fcc0-d276-11eb-a888-3141ef71e4d9" data-business-id="9f8ab984-54c1-4ccc-b774-8d8c206bdce5" href="none" data-finding-id="476566751" repeated-comment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+        && gem install bundler --no-document \
+        && gem install solargraph --no-document" \
+    && echo '[[ -s "$HOME/.rvm/scripts/rvm" ]] && source "$HOME/.rvm/scripts/rvm" # Load RVM into a shell session *as a function*' >> /home/gitpod/.bashrc.d/70-ruby
+RUN echo "rvm_gems_path=/workspace/.rvm" > ~/.rvmrc
+
+### Rust ###
+LABEL dazzle/layer=lang-rust
+LABEL dazzle/test=tests/lang-rust.yaml
+USER gitpod
+RUN cp /home/gitpod/.profile /home/gitpod/.profile_orig && \
```
Comment (alexander-a-chernyshev) [inline full/Dockerfile.arm64:198]: <details>
<summary>:no_entry_sign: :repeat: <strong>This RUN command is calling sudo.</strong> (click for details)</summary>
First occurrence of this finding in this pull request can be found <a href="https://github.com/trilogy-group/workspace-images/pull/1#discussion_null">here</a></details>

[//]: # (<a id="18c0e240-d276-11eb-a888-3141ef71e4d9" data-business-id="9f8ab984-54c1-4ccc-b774-8d8c206bdce5" href="none" data-finding-id="476566751" repeated-comment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+    && .cargo/bin/rustup completions bash | sudo tee /etc/bash_completion.d/rustup.bash-completion > /dev/null \
+    && .cargo/bin/rustup completions bash cargo | sudo tee /etc/bash_completion.d/rustup.cargo-bash-completion > /dev/null \
+    && grep -v -F -x -f /home/gitpod/.profile_orig /home/gitpod/.profile > /home/gitpod/.bashrc.d/80-rust
+ENV PATH=$PATH:$HOME/.cargo/bin
+
+RUN bash -lc "cargo install cargo-watch cargo-edit cargo-tree"
+
+### Install additional components ###
+USER gitpod
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-aarch64.zip" -o "awscliv2.zip" && \
```
Comment (alexander-a-chernyshev) [inline full/Dockerfile.arm64:214]: <details>
<summary>:no_entry_sign: :repeat: <strong>This RUN command is calling sudo.</strong> (click for details)</summary>
First occurrence of this finding in this pull request can be found <a href="https://github.com/trilogy-group/workspace-images/pull/1#discussion_null">here</a></details>

[//]: # (<a id="1b30a9c0-d276-11eb-a888-3141ef71e4d9" data-business-id="9f8ab984-54c1-4ccc-b774-8d8c206bdce5" href="none" data-finding-id="476566751" repeated-comment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+    unzip awscliv2.zip && \
+    sudo ./aws/install
+RUN curl --silent --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_Linux_arm64.tar.gz" | tar xz -C /tmp && \
```
Comment (alexander-a-chernyshev) [inline full/Dockerfile.arm64:217]: <details>
<summary>:no_entry_sign: :repeat: <strong>This RUN command is calling sudo.</strong> (click for details)</summary>
First occurrence of this finding in this pull request can be found <a href="https://github.com/trilogy-group/workspace-images/pull/1#discussion_null">here</a></details>

[//]: # (<a id="1eb78690-d276-11eb-a888-3141ef71e4d9" data-business-id="9f8ab984-54c1-4ccc-b774-8d8c206bdce5" href="none" data-finding-id="476566751" repeated-comment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+    sudo mv /tmp/eksctl /usr/local/bin
+RUN sudo apt-get update && \
```
Comment (alexander-a-chernyshev) [inline full/Dockerfile.arm64:219]: <details>
<summary>:no_entry_sign: :repeat: <strong>This RUN command is calling sudo.</strong> (click for details)</summary>
First occurrence of this finding in this pull request can be found <a href="https://github.com/trilogy-group/workspace-images/pull/1#discussion_null">here</a></details>

[//]: # (<a id="21a816d0-d276-11eb-a888-3141ef71e4d9" data-business-id="9f8ab984-54c1-4ccc-b774-8d8c206bdce5" href="none" data-finding-id="476566751" repeated-comment codereview-uuid-reserved-tag />)

**Review by alexander-a-chernyshev — COMMENTED** [review]
Hi, I am the IaC expert who performed the review of this Pull Request. We have well-defined [coding and review standards](https://confluence.devfactory.com/display/CodeReview/Findings), and you will find below suggestions to improve the code quality, based on these guidelines.
My additional comments are in the Notes section. I hope you find them useful.

I raised many findings on this Pull Request and shared below partial list of suggestions to improve the code quality, kindly consider the suggestions and similar patterns on the code while updating the Pull Request, then the Review team will perform another full review once it is updated to provide more suggestions.


<table>
  <tr>
    <td>:no_entry_sign:<b>Severes:</b>  11</td>
    <td>:exclamation:<b>Reds:</b>  0</td>
    <td>:warning:<b>Yellows:</b>  0</td>
    <td>:white_check_mark:<b>Greens:</b>  0</td>
  </tr>
</table>

<details><summary>More information</summary>
  <table>
    <tr>
      <td>:no_entry_sign:</td>
      <td><b>Severe Findings</b> point out serious violations which can affect stability, performance, maintainability and therefore should be fixed.
        <ul>
          <li>Please note that the merge might have been blocked until all Severe Findings are fixed.</li>
          <li>While it is highly recommended to implement all findings only Severe Findings might have been blocking the merge</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>:exclamation:</td>
      <td><b>Red Findings</b> represent potential bugs and it is highly recommended to implement them.<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td><b>Yellows</b> are suggestions to improve the code, warnings and tips<br/>
      Not required for the approval of the PR, and should not be used for FTAR calculation.
      </td>
    </tr>
    <tr>
      <td>:white_check_mark:</td>
      <td><b>Green Findings</b> highlight good ideas and solutions</td>
    </tr>
    <tr>
      <td>:repeat:</td>
      <td><b>Repeated</b> instances of the same findings are counted just once and flagged with the repeat icon</td>
    </tr>
  </table>

  Feel free to reply to this comment, (make sure you `@tag` us) so we can start a conversation. The goal here is to learn and grow together and make our products better.

  > NOTE: This review **does not** replace your team's review._
</details>
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR decreased your FTAR from 100.00 % to 75.00 %, so we recommend studying raised finding to improve the quality of written code for next upcoming PRs.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>75.00 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>4</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>3</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>1</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Comment by alexander-a-chernyshev** [comment]
Hi, I'm Alexander Chernyshev from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely are you to recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/workspace-images/pull/1&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

**Comment by vdefeo-xo** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

Please consider using [Analyzer UI](https://confluence.devfactory.com/display/CodeReview/Analyzer+UI) extension to catch issues in pull requests.

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: full/Dockerfile [hunk 10]
```diff
@@ -207,6 +208,20 @@ ENV PATH=$PATH:$HOME/.cargo/bin
 
 RUN bash -lc "cargo install cargo-watch cargo-edit cargo-tree"
 
+### Install additional components ###
+USER gitpod
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
```
[the remark this answers is not in the document]
Comment (vdefeo-xo) [reply full/Dockerfile:213]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="cc17f670-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile (hunk 2 above)
Comment (vdefeo-xo) [reply full/Dockerfile:215]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="cd7f75b0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile [hunk 11]
```diff
@@ -207,6 +208,20 @@ ENV PATH=$PATH:$HOME/.cargo/bin
...
+    unzip awscliv2.zip && sudo ./aws/install
+RUN curl --silent --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_Linux_amd64.tar.gz" | tar xz -C /tmp && \
+    sudo mv /tmp/eksctl /usr/local/bin
+RUN sudo apt-get update && \
```
[the remark this answers is not in the document]
Comment (vdefeo-xo) [reply full/Dockerfile:217]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="cf037da0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 (hunk 3 above)
Comment (vdefeo-xo) [reply full/Dockerfile.arm64:69]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="d0630da0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 (hunk 4 above)
Comment (vdefeo-xo) [reply full/Dockerfile.arm64:85]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="d194d6e0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 (hunk 5 above)
Comment (vdefeo-xo) [reply full/Dockerfile.arm64:157]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="d3219160-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 12]
```diff
@@ -0,0 +1,268 @@
...
+ENV PATH=$PATH:/home/gitpod/.nvm/versions/node/v${NODE_VERSION}/bin
+
+### Python ###
+LABEL dazzle/layer=lang-python
+LABEL dazzle/test=tests/lang-python.yaml
+USER gitpod
+RUN sudo install-packages python3-pip
+
+ENV PATH=$HOME/.pyenv/bin:$HOME/.pyenv/shims:$PATH
+RUN curl -fsSL https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash \
```
[the remark this answers is not in the document]
Comment (vdefeo-xo) [reply full/Dockerfile.arm64:160]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="d54ab2f0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 (hunk 6 above)
Comment (vdefeo-xo) [reply full/Dockerfile.arm64:198]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="d68cf6f0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 (hunk 7 above)
Comment (vdefeo-xo) [reply full/Dockerfile.arm64:214]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="d7eed0e0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 (hunk 8 above)
Comment (vdefeo-xo) [reply full/Dockerfile.arm64:217]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="d93a63b0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 (hunk 9 above)
Comment (vdefeo-xo) [reply full/Dockerfile.arm64:219]: <details>
<summary>:no_entry_sign: <strong>This RUN command is calling sudo.</strong> (unfixed, click for details)</summary>

This finding is still pending without a fix, technical justification or PCA approval.
</details>

[//]: # (<a id="daa3b7b0-d2b8-11eb-b193-1735994bb092" data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" href="none" data-finding-id="476566751" blocking-flow-recomment codereview-uuid-reserved-tag />)

File: full/Dockerfile.arm64 [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+#    && apt install /tmp/dive.deb \
+#    && rm /tmp/dive.deb
+COPY dive-arm64.zip .
+RUN unzip dive-arm64.zip -d /usr/bin && chmod +x /usr/bin/dive
+
+### Prologue (built across all layers) ###
+LABEL dazzle/layer=dazzle-prologue
+LABEL dazzle/test=tests/prologue.yaml
+USER root
+#RUN curl -o /usr/bin/dazzle-util -fsSL https://github.com/csweichel/dazzle/releases/download/v0.1.0/dazzle-util_0.1.0_Linux_arm64 \
```
Comment (vdefeo-xo) [inline full/Dockerfile.arm64:254]: <details>
<summary>:warning: <strong>Commented-out Code is Committed.</strong> (click for details)</summary>

This pull request contains commented code which is not recommended when a version control repository (e.g., GitHub) is used. It is recommended to delete all commented instructions before committing to VCS.

**[More details](https://confluence.devfactory.com/x/yAV7HQ)**

## The improvement suggestion in this case
Consider avoiding committing commented code.
</details>

[//]: # (<a id="1fd69ee5-d0be-45b6-b09b-ea7989fb4e40" data-finding-id="494601672" href="none" codereview-uuid-reserved-tag data-business-id="b5d55598-2af3-4612-9fe2-ef549f2ff47f" />)

**Review by vdefeo-xo — COMMENTED** [review]
Hi, I am the IaC expert who performed the review of this Pull Request. We have well-defined [coding and review standards](https://confluence.devfactory.com/display/CodeReview/Findings), and you will find below suggestions to improve the code quality, based on these guidelines.
My additional comments are in the Notes section. I hope you find them useful.

<table>
  <tr>
    <td>:no_entry_sign:<b>Severes:</b>  11</td>
    <td>:exclamation:<b>Reds:</b>  0</td>
    <td>:warning:<b>Yellows:</b>  1</td>
    <td>:white_check_mark:<b>Greens:</b>  0</td>
  </tr>
</table>

<details><summary>More information</summary>
  <table>
    <tr>
      <td>:no_entry_sign:</td>
      <td><b>Severe Findings</b> point out serious violations which can affect stability, performance, maintainability and therefore should be fixed.
        <ul>
          <li>Please note that the merge might have been blocked until all Severe Findings are fixed.</li>
          <li>While it is highly recommended to implement all findings only Severe Findings might have been blocking the merge</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>:exclamation:</td>
      <td><b>Red Findings</b> represent potential bugs and it is highly recommended to implement them.<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td><b>Yellows</b> are suggestions to improve the code, warnings and tips<br/>
      Not required for the approval of the PR, and should not be used for FTAR calculation.
      </td>
    </tr>
    <tr>
      <td>:white_check_mark:</td>
      <td><b>Green Findings</b> highlight good ideas and solutions</td>
    </tr>
    <tr>
      <td>:repeat:</td>
      <td><b>Repeated</b> instances of the same findings are counted just once and flagged with the repeat icon</td>
    </tr>
  </table>

  Feel free to reply to this comment, (make sure you `@tag` us) so we can start a conversation. The goal here is to learn and grow together and make our products better.

  > NOTE: This review **does not** replace your team's review._
</details>
<details><summary>Pull Request Blockers</summary>
<table>
  <tr>
    <td><b>Fixed blockers:</b> 0</td>
    <td><b>Delegated blockers to PCA:</b> 0</td>
    <td><b>Unfixed blockers:</b> 11</td>
    <td><b>Deleted findings:</b> 0</td>
  </tr>
</table>
</details>
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 66.67 % to 75.00 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>75.00 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>4</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>3</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>1</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Review by pavan-tri — APPROVED** [review]
LGTM

**Pull request merged**